mailpoet/tests/acceptance/Automation/CreateEmailAutomationAndWalkThroughCest.php

```diff
@@ -0,0 +1,110 @@
+    $container->get(FeatureFlagsController::class)->set(FeaturesController::AUTOMATION, true);
+    $container->get(SettingsController::class)->set('cron_trigger.method', 'WordPress');
+  }
+
+  public function createEmailWorkflowAndReceiveAnAutomatedEmail(\AcceptanceTester $i) {
+    $i->wantTo('Create a workflow to send an email after a user subscribed');
+    $i->login();
+
+    $i->amOnMailpoetPage('Automation');
+    $i->see('Automations');
+    $i->waitForText('Scale your business with advanced automations');
+    $i->dontSee('Simple welcome email');
+    $i->dontSee('Active');
+    $i->dontSee('Entered');
+
+    $i->click('New automation');
+    $i->see('Choose your automation template');
+    $i->click('Simple welcome email');
+
+    $i->waitForText('Draft');
+    $i->click('Trigger');
+    $i->fillField('When someone subscribers to the following list(s):', 'Newsletter mailing list');
+    $i->click('Delay');
+    $i->fillField('Wait for', '5');
+
+    $i->click('Send email');
+    $i->fillField('“From” name','From');
+    $i->fillField('“From” email address','test@mailpoet.com');
+    $i->fillField('Subject','Automation-Test-Subject');
+
+    $i->click('Design email');
+    $i->waitForText('Newsletters');
+    $i->click('Newsletters');
+    $i->click('button[data-automation-id="select_template_0"]');
+    $i->waitForText('Design');
+    $i->click('Save and continue');
+
+    $i->waitForText('Draft');
+
+    $i->click('Send email');
+    $i->click('Reply to');
+    $i->fillField('“Reply to” name', 'Reply');
+    $i->fillField('“Reply to” email address', 'reply@mailpoet.com');
+
+    $i->click('Activate');
+
+    // Check workflow is activated
+    $i->waitForText('Update');
+    $i->see('Active');
+
+    $i->click('button[aria-label="Options"]');
+    $i->click('View all automations');
+    $i->waitForText('Name');
+    $i->see('Simple welcome email');
+    $i->see('Active');
+    $i->see('0 Entered');
+    $i->dontSeeInDatabase('mp_actionscheduler_actions', ['hook' => 'mailpoet/automation/workflow/step']);
+
+    $i->wantTo('Check a new subscriber gets the automation email.');
+    $i->amOnPage('/wp-admin/admin.php?page=mailpoet-subscribers#/new');
+    $i->fillField('#field_email', 'test@mailpoet.com');
+    $i->fillField('#field_first_name', 'automation-tester-firstname');
+    $i->selectOptionInSelect2('Newsletter mailing list');
+    $i->click('Save');
+
+    $i->amOnMailpoetPage('Automation');
+    $i->seeInDatabase('mp_actionscheduler_actions', ['hook' => 'mailpoet/automation/workflow/step', 'status' => 'pending']);
+    $i->waitForText('Simple welcome email');
+    $i->see('1 Entered');
+    $i->see('1 Processing');
+    $i->see('0 Exited');
+    $i->amOnMailboxAppPage();
+    $i->see('Inbox (0)');
+
+    // Jump the waiting time by scheduling the delay action to now.
+    $i->triggerAutomationActionScheduler();
+    $i->triggerAutomationActionScheduler();
```

Comment:
Why is it necessary to call this twice? It's the number of steps in the workflow,  right? Maybe we could add some comments like:

```php
    $i->triggerAutomationActionScheduler(); // delay step
    $i->triggerAutomationActionScheduler(); // send email step
```

 Do we want/need to do any assertions in between those two steps?

Reply:
Addressed in [9b286b3](https://github.com/mailpoet/mailpoet/pull/4434/commits/9b286b3d949a2ee01d3b921e776b81ab94fa5220)

Reply:
> Do we want/need to do any assertions in between those two steps?

I don't think so. To me the need to run it twice is already to much "looking into the black box". On the level of acceptance test, I don't think it makes sense to dive so deep for assertions.

Now I wonder about the database lookup I actually do [to check if we have a pending task in the Action Scheduler](https://github.com/mailpoet/mailpoet/pull/4434/files/8e50e4e608127de9aa9a54b3f001da8e9df371b6#diff-68667e9409836537b4425bd4f07c6aa5f5a7f5d26e4955eb233dbe196f5343f0R86), but I guess, we can leave it as it does not do any harm :see_no_evil: